lib/user.rb

```diff
@@ -0,0 +1,39 @@
+require 'pg'
+
+class User
+  attr_reader :id, :name, :username
+
+  def initialize(id, name, username, email, password)
+    @id = id
+    @name = name
+    @username = username
+    @email = email
+    @password = password
+  end
+
+  def self.add(name, username, email, password)
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
```

Comment:
here it appears again, in step 14 of the bookmark-manager there is guidance on extracting the database logic to its own object [ Extracting the database connection logic to an object](https://github.com/makersacademy/course/blob/master/bookmark_manager/walkthroughs/14.md).

Reply:
amazing, thank you!!